app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt

```diff
@@ -45,73 +43,44 @@ import com.wire.kalium.logic.data.id.ConversationId
-                    MutingOptionsSheetContent(
-                        mutingConversationState = mutingConversationState,
-                        onItemClick = { conversationId, mutedStatus ->
-                            viewModel.muteConversation(conversationId, mutedStatus)
-                            conversationState.modalBottomSheetContentState.value.updateCurrentEditingMutedStatus(mutedStatus)
-                            conversationState.toggleEditMutedSetting(false)
-                        },
-                        onBackClick = {
-                            mutingConversationState.closeMutedStatusSheetContent()
-                            conversationState.modalBottomSheetContentState.value.updateCurrentEditingMutedStatus(
-                                mutingConversationState.mutedStatus
-                            )
-                            conversationState.toggleEditMutedSetting(false)
-                        }
-                    )
-                }
-                false -> {
-                    ConversationSheetContent(
-                        modalBottomSheetContentState = conversationState.modalBottomSheetContentState.value,
-                        notificationsOptionsItem = NotificationsOptionsItem(
-                            muteConversationAction = {
-                                mutingConversationState.openMutedStatusSheetContent(
-                                    conversationState.modalBottomSheetContentState.value.conversationId,
-                                    conversationState.modalBottomSheetContentState.value.mutedStatus
-                                )
-                                // here we trigger a sheet content change, enabling muted settings toggle
-                                conversationState.toggleEditMutedSetting(true)
-                            },
-                            mutedStatus = mutingConversationState.mutedStatus
-                        ),
-                        addConversationToFavourites = { viewModel.addConversationToFavourites("someId") },
-                        moveConversationToFolder = { viewModel.moveConversationToFolder("someId") },
-                        moveConversationToArchive = { viewModel.moveConversationToArchive("someId") },
-                        clearConversationContent = { viewModel.clearConversationContent("someId") },
-                        blockUser = { viewModel.blockUser("someId") },
-                        leaveGroup = { viewModel.leaveGroup("someId") }
-                    )
-                }
+    LaunchedEffect(conversationState.conversationSheetContent) {
+        conversationState.conversationSheetContent?.let { conversationSheetContent ->
+            onHomeBottomSheetContentChange {
+                ConversationSheetContent(
+                    conversationSheetContent = conversationSheetContent,
+                    onMutingConversationStatusChange = conversationState::muteConversation,
```

Comment:
Just curious, these changes will solve the recomoposition issue, similar of what we had in the past with conversations list weeks ago ?

Reply:
I mean the function ref vs the lambda block? 🤔

Reply:
honestly no idea if that helps here :P but just to make sure wherever possible I use function reference instead of the lambda just to make sure that it will not be recomposed or anything